packages/odf/modals/capacity-autoscaling/capacity-autoscaling-modal.tsx

```diff
@@ -0,0 +1,404 @@
+  ModalVariant,
+} from '@patternfly/react-core';
+import { InProgressIcon, RunningIcon } from '@patternfly/react-icons';
+import './capacity-autoscaling-modal.scss';
+
+const getFormattedTime = (timestamp: string) =>
+  dateTimeFormatter.format(new Date(timestamp));
+
+type CapacityAutoscalingStatusProps = {
+  currentStatus: string;
+  lastExpansion?: StorageAutoScalerKind['status']['lastExpansion'];
+};
+
+const CapacityAutoscalingCurrentStatus: React.FC<
+  CapacityAutoscalingStatusProps
+> = ({ currentStatus, lastExpansion }) => {
+  const { t } = useCustomTranslation();
+
+  let StatusIcon:
+    | React.ComponentClass<SVGIconProps>
+    | React.FC<ColoredIconProps>;
+  let statusText: string;
+  switch (currentStatus) {
+    case CapacityAutoscalingStatus.NotStarted:
+      StatusIcon = RunningIcon;
+      statusText = t('Actively monitoring, waiting for trigger');
+      break;
+    case CapacityAutoscalingStatus.InProgress:
+      StatusIcon = InProgressIcon;
+      statusText = t('Scaling is in progress');
+      if (lastExpansion?.startTime) {
+        statusText += `, ${t('started at ')} ${getFormattedTime(lastExpansion?.startTime)}`;
+      }
+      break;
+    case CapacityAutoscalingStatus.Succeeded:
+      StatusIcon = GreenCheckCircleIcon;
+      statusText = t('Completed');
+      if (lastExpansion?.completionTime) {
+        statusText += `, ${getFormattedTime(lastExpansion?.completionTime)}`;
+      }
+      break;
+    case CapacityAutoscalingStatus.Failed:
+      StatusIcon = RedExclamationCircleIcon;
+      statusText = t('Failed');
+      if (lastExpansion?.startTime) {
+        statusText += `, ${t('started at ')} ${getFormattedTime(lastExpansion?.startTime)}`;
+      }
+      break;
+    default:
+      StatusIcon = GrayUnknownIcon;
+      statusText = t('Unknown');
+  }
+
+  return (
+    <div className="pf-v5-u-mt-sm">
+      <span className="pf-v5-u-font-family-heading pf-v5-u-mr-sm">
+        {t('Current status:')}
+      </span>
+      <StatusIcon className="pf-v5-u-mr-sm" />
+      {statusText}
+    </div>
+  );
+};
+
+type CapacityAutoscalingModalProps = {
+  storageCluster: StorageClusterKind;
+} & CommonModalProps;
+
+const CapacityAutoscalingModal: React.FC<CapacityAutoscalingModalProps> = ({
+  storageCluster,
+  closeModal,
+  isOpen,
+}) => {
+  const { t } = useCustomTranslation();
+  const [enable, setEnable] = React.useState(false);
+  const [capacityLimit, setCapacityLimit] = React.useState<string>(
+    getCapacityAutoScalingDefaultLimit()
+  );
+  const [validation, setValidation] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMessage, setError] = React.useState<Error>(null);
+  const [
+    storageAutoScalers,
+    storageAutoScalerLoaded,
+    storageAutoScalerLoadError,
+  ] = useSafeK8sWatchResource<StorageAutoScalerKind[]>((ns: string) => ({
+    kind: referenceForModel(StorageAutoScalerModel),
+    isList: true,
+    namespace: ns,
+  }));
```

Comment:
`useSafeK8sWatchResource` uses ODF operator install namespace, what you need here is StorageCluster namespace...
```suggestion
  ] = useK8sWatchResource<StorageAutoScalerKind[]>({
    kind: referenceForModel(StorageAutoScalerModel),
    isList: true,
    namespace: getNamespace(storageCluster),
  }));
```